src/pages/index.js

```diff
@@ -85,125 +130,147 @@ const cardUrlInput = addCardFormElement.querySelector(".modal__input_type_url");
-  // modalImage.src = cardData.link;
-  // modalImage.alt = cardData.name;
-  // modalTitle.textContent = cardData.name;
   imagePreviewPopup.open(cardData);
 }
 
-// function renderCard(cardData) {
-//   const cardElement = createCard(cardData);
-//   cardSection.addItem(cardElement);
-// }
-
-// function getCardElement(cardData) {
-//   const cardElement = cardTemplate.cloneNode(true);
-//   const cardImageElement = cardElement.querySelector(".card__image");
-//   const cardTitleElement = cardElement.querySelector(
-//     ".card__description-title"
-//   );
-//   const likeButton = cardElement.querySelector(".card__like-button");
-
-//   likeButton.addEventListener("click", () => {
-//     likeButton.classList.toggle("card__like-button_active");
-//   });
-
-//   const trashButton = cardElement.querySelector(".card__trash-button");
-//   trashButton.addEventListener("click", () => {
-//     cardElement.remove();
-//   });
-
-//   cardImageElement.addEventListener("click", () => {
-//     modalImage.src = cardData.link;
-//     modalImage.alt = cardData.name;
-//     modalTitle.textContent = cardData.name;
-//     imagePreviewPopup.open();
-//   });
-
-//   cardImageElement.src = cardData.link;
-//   cardTitleElement.textContent = cardData.name;
-//   cardImageElement.alt = cardData.name;
-//   return cardElement;
-// }
-
 function handleProfileEditSubmit(data) {
-  profileTitle.textContent = profileTitleInput.value;
-  profileDescription.textContent = profileDescriptionInput.value;
-  editProfilePopup.close();
+  // data is an object with name and subtitle
+  editProfilePopup.renderLoading(true);
+  api
+    .updateProfileInfo(data)
+    .then(() => {
+      userInfo.setUserInfo({ title: data.name, description: data.subtitle });
+      // profileTitle.textContent = profileTitleInput.value;
+      // profileDescription.textContent = profileDescriptionInput.value;
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err}, Could not update user info!`);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
 }
 
 function handleAddCardSubmit(inputValues) {
-  // const name = cardTitleInput.value;
-  // const link = cardUrlInput.value;
-  cardSection.addItem({ name: inputValues.title, link: inputValues.url });
-  addCardFormElement.reset();
-  addCardPopup.close();
-  // cardTitleInput.value = "";
-  // cardUrlInput.value = "";
-  addCardFormValidator.disableButton();
+  // addCardFormElement.reset();
+  // addCardPopup.close();
+  // addCardFormValidator.disableButton();
+  addCardPopup.renderLoading(true);
+  api
+    .addCard({ name: inputValues.title, link: inputValues.url })
+    .then((newCardData) => {
+      cardSection.addItem(newCardData);
+      addCardFormElement.reset();
+      addCardFormValidator.resetValidation();
+      addCardPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
```

Comment:
`      editProfilePopup.renderLoading(false);
`

Please, use the correct instance in `finally` to stop the loading effect


`addCardPopup` shoul be used here